kusto/query_builder.go

```diff
@@ -506,6 +435,166 @@ func NewStmt(query stringConstant, options ...StmtOption) Stmt {
+	return s.NormalizeName(query, false)
+}
+
+// AddColumn - Given a variable with a value 'MyColumn', AddColumn(arg) will produce the following string
+// '["MyColumn"]' and will append the following string to Stmt
+func (s Stmt) AddColumn(query string) Stmt {
+	return s.NormalizeName(query, false)
+}
+
+// AddFunction - Given a variable with a value 'MyFunction', AddFunction(arg) will produce the following string
+// '["MyFunction"]' and will append the following string to Stmt
+func (s Stmt) AddFunction(query string) Stmt {
+	return s.NormalizeName(query, false)
+}
+
+// NormalizeName normalizes a string in order to be used safely in the engine - given "query" will produce [\"query\"].
+func (s Stmt) NormalizeName(query string, forceNormalization bool) Stmt {
+	if query == "" {
+		return s
+	}
+	if !forceNormalization && !RequiresQuoting(query) {
+		s.queryStr = s.queryStr + query
+		return s
+	}
+
+	s.queryStr = s.queryStr + "[" + AddQuotedString(query, false) + "]"
+	return s
+}
+
+// RequiresQuoting checks whether a given string is an identifier
+func RequiresQuoting(query string) bool {
+	if query == "" {
+		return true
+	}
+	if !unicode.IsLetter(rune(query[0])) && rune(query[0]) != '_' {
+		return true
+	}
+	for _, c := range query {
+		if !(unicode.IsLetter(c) || unicode.IsDigit(c) || c == '_') {
+			return true
+		}
+	}
+	return false
+}
+
+// AddQuotedString escapes a string to be safely added to a stmt
+func AddQuotedString(value string, hidden bool) string {
+	if value == "" {
+		return value
+	}
+
+	var literal strings.Builder
+
+	if hidden {
+		literal.WriteString("h")
+	}
+	literal.WriteString("\"")
+	for _, c := range value {
+		switch c {
+		case '\'':
+			literal.WriteString("\\'")
+
+		case '"':
+			literal.WriteString("\\\"")
+
+		case '\\':
+			literal.WriteString("\\\\")
+
+		case '\x00':
+			literal.WriteString("\\0")
+
+		case '\a':
+			literal.WriteString("\\a")
+
+		case '\b':
+			literal.WriteString("\\b")
+
+		case '\f':
+			literal.WriteString("\\f")
+
+		case '\n':
+			literal.WriteString("\\n")
+
+		case '\r':
+			literal.WriteString("\\r")
+
+		case '\t':
+			literal.WriteString("\\t")
+
+		case '\v':
+			literal.WriteString("\\v")
+
+		default:
+			if !ShouldBeEscaped(c) {
+				literal.WriteString(string(c))
+			} else {
+				literal.WriteString(fmt.Sprintf("\\u%04x", c))
+			}
+
+		}
+	}
+	literal.WriteString("\"")
+
+	return literal.String()
+}
+
+// ShouldBeEscaped Checks whether a rune should be escaped or not based on it's type.
+func ShouldBeEscaped(c int32) bool {
+	if c <= unicode.MaxLatin1 {
+		return unicode.IsControl(c)
+	}
+	return true
+}
+
+// AddInt will add an int as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddInt(query int) Stmt {
+	return s.addBase(types.Int, query)
+}
+
+// AddFloat32 will add a Float32 as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddFloat32(query float32) Stmt {
```

Comment:
This (and 64) is wrong.

The type should be types.Float.

Reply:
you mean in the type passed into addbase? we've built the other function using types.Decimal - and there is also no types.Float 
![image](https://user-images.githubusercontent.com/100577889/206438789-49bcb1e4-acef-43b8-a25e-09e45596c384.png)